pd/src/pending_block.rs

```diff
@@ -16,14 +17,21 @@ pub struct PendingBlock {
     pub note_commitment_tree: NoteCommitmentTree,
     /// Stores note commitments for convienience when updating the NCT.
     pub notes: BTreeMap<note::Commitment, PositionedNoteData>,
+    /// Notes that are undelegation outputs in this block (not to be committed to the NCT yet).
+    pub undelegation_notes: BTreeMap<note::Commitment, QuarantinedNoteData>,
     /// Nullifiers that were spent in this block.
     pub spent_nullifiers: BTreeSet<Nullifier>,
+    /// Nullifiers that were associated with undelegation transactions in this block, associated
+    /// with their unbonding epoch and validator identity key.
+    pub quarantined_nullifiers: BTreeMap<Nullifier, u64>,
     /// Records any updates to the token supply of some asset that happened in this block.
     pub supply_updates: BTreeMap<asset::Id, (asset::Denom, u64)>,
     /// Indicates the epoch the block belongs to.
     pub epoch: Option<Epoch>,
     /// Indicates the duration in blocks of each epoch.
     pub epoch_duration: u64,
+    /// Indicates the number of epochs before undelegated stake is unbonded.
+    pub unbonding_epochs: u64,
```

Comment:
At this point we've replicated two fields from the chain parameters into the pending block -- should we just have it store the whole set of parameters?